crates/oxc_cli/src/walk.rs

```diff
@@ -3,26 +3,30 @@ use std::path::Path;
 use ignore::{DirEntry, WalkBuilder};
 use oxc_ast::VALID_EXTENSIONS;
 
-use crate::CliOptions;
+use crate::{git::Git, CliOptions};
 
 pub struct Walk {
     inner: ignore::Walk,
+    git: Git,
 }
 
 impl Walk {
-    pub fn new<P: AsRef<Path>>(path: P, options: &CliOptions) -> Self {
+    pub fn new<P: AsRef<Path> + Copy>(path: P, options: &CliOptions) -> Self {
+        let git = Git::new(path);
         let mut inner = WalkBuilder::new(path);
         if !options.no_ignore {
             inner.add_custom_ignore_filename(&options.ignore_path);
         }
         let inner = inner.ignore(false).git_global(false).build();
-        Self { inner }
+        Self { inner, git }
     }
 
     pub fn iter(self) -> impl Iterator<Item = Box<Path>> {
         self.inner
             .filter_map(Result::ok)
-            .filter(Self::is_wanted_entry)
+            .filter(move |entry| {
+                Self::is_wanted_entry(entry) && !self.git.is_uncommitted(entry.path())
```

Comment:
hmmm, I thought we should just bail the whole `--fix` instead of checking whether each file is uncommited or not, we should follow how Clippy does it.